octoprint_mrbeam/service/profile/profile.py

```diff
@@ -0,0 +1,482 @@
+        default = settings().get(["profiles", self._id, "default"])
+        return default is None or default == self.DEFAULT_PROFILE_ID or not self.exists(self.DEFAULT_PROFILE_ID)
+
+    @property
+    def profile_count(self):
+        return len(self._load_all_identifiers())
+
+    @property
+    def last_modified(self):
+        dates = [os.stat(self._folder).st_mtime]
+        dates += [entry.stat().st_mtime for entry in scandir(self._folder) if entry.name.endswith(".profile")]
+        return max(dates)
+
+    def get_default(self):
+        default = settings().get(["profiles", self._id, "default"])
+        if default is not None and self.exists(default):
+            profile = self.get(default)
+            if profile is not None:
+                return profile
+            else:
+                self._logger.warn("Default profile {} is invalid, falling back to built-in defaults".format(default))
+
+        return copy.deepcopy(self._default)
+
+    def set_default(self, identifier):
+        all_identifiers = self._load_all_identifiers().keys()
+        if identifier is not None and not identifier in all_identifiers:
+            return
+
+        settings().set(["profiles", self._id, "default"], identifier, force=True)
+        settings().save()
+
+    def get_current_or_default(self):
+        if self._current is not None:
+            return self._current
+        else:
+            return self.get_default()
+
+    def get_current(self):
+        return self._current
+
+    def exists(self, identifier):
+        if identifier is None:
+            return False
+        else:
+            path = self._get_profile_path(identifier)
+            return os.path.exists(path) and os.path.isfile(path)
+
+    def _load_all(self):
+        all_identifiers = self._load_all_identifiers()
+        results = dict()
+        for identifier, path in all_identifiers.items():
+            try:
+                profile = self._load_from_path(path)
+            except InvalidProfileError:
+                self._logger.warn("Profile {} is invalid, skipping".format(identifier))
+                continue
+
+            if profile is None:
+                continue
+
+            results[identifier] = dict_merge(self._default, profile)
+        return results
+
+    def _load_all_identifiers(self):
+        results = dict()
+        for entry in scandir(self._folder):
+            if is_hidden_path(entry.name) or not entry.name.endswith(".profile"):
+                continue
+
+            if not entry.is_file():
+                continue
+
+            identifier = entry.name[:-len(".profile")]
+            results[identifier] = entry.path
+        return results
+
+    def _load_from_path(self, path):
+        if not os.path.exists(path) or not os.path.isfile(path):
+            return None
+
+        import yaml
+        with open(path) as f:
+            profile = yaml.safe_load(f)
+
+        if profile is None or not isinstance(profile, dict):
```

Comment:
```suggestion
        if not isinstance(profile, dict):

```

Reply:
This is not my implementation but OctoPrint's. We are using their class as a parent class with minor changes to fir our purpose